packages/flyout/src/presenters/FlyoutPresenter.js

```diff
@@ -48,16 +49,39 @@ export default function FlyoutPresenter(props) {
           resolvedRoles
         );
 
+        const { className } = otherProps;
+        const classNameArray = className
+          ? className.split(" ").map(name => name.trim())
+          : [];
+
         return (
-          <div className={css(styles.flyoutWrapper)} ref={refWrapper}>
-            <div className={css(styles.flyoutAction)} ref={refAction}>
+          <div
+            className={cx(css(styles.flyoutWrapper), className)}
+            ref={refWrapper}
+          >
+            <div
+              className={cx(
+                css(styles.flyoutAction),
+                classNameArray.map(name => `${name}-action`)
+              )}
+              ref={refAction}
+            >
               {children}
             </div>
-            <div className={css(styles.flyoutContainer)} style={containerStyle}>
+            <div
+              className={cx(
+                css(styles.flyoutContainer),
+                classNameArray.map(name => `${name}-container`)
+              )}
+              style={containerStyle}
+            >
               <PointerWrapperPresenter
                 innerRef={refPointer}
                 anchorPoint={anchorPoint}
                 style={pointerStyle}
+                className={cx(
```

Comment:
Hmm, my thinking is: Parent should pass the final className that should be added to its child components, and for any component, it will just add what props.className that it received from its parent, to its outer wrapper. then add --sufix to its children. 🤔

Reply:
In this case, the FlyoutPresenter should pass the final className for Pointer, like 'myclass-container-pointer' to Pointer, and the Pointer will just append what it receives from prop to its outer DIV. and then for Pointer's children, it can add other classnames with suffix...

Reply:
I am going to take a look at this some more, we can discuss this more after I look at the component a little more.

Reply:
@garyps I think this is similar to what you did but some minor changes to the naming convention: https://github.com/Autodesk/hig/pull/1979

Reply:
Checked #1979 and i got your points of the way on adding classnames, sure, think we can close this PR and just merge #1979 ?